app/src/main/java/ch/epfl/sweng/favors/database/storage/Storage.java

```diff
@@ -110,20 +125,74 @@ public String uploadImage(StorageReference storageReference, Context context, Ur
     }
 
     @Override
-    public void displayImage(ObservableField<String> pictureRef, ImageView imageView) {
-        if(pictureRef != null && pictureRef.get() != null){
-            StorageReference ref = FirebaseStorage.getInstance().getReference().child("images/"+ pictureRef.get());
+    public void displayImage(ObservableField<String> pictureRef, ImageView imageView, String category) {
+
+        if(Storage.checkStoragePath(category) && pictureRef != null && pictureRef.get() != null){
+            String path = category.toLowerCase()+"/";
+            StorageReference ref = FirebaseStorage.getInstance().getReference().child(path+ pictureRef.get());
             view = imageView;
             ref.getBytes(MAX_BYTE_SIZE).addOnSuccessListener(byteSuccessListener);
 
         }
     }
 
     @Override
-    public void deleteImageFromStorage(ObservableField<String> pictureRef) {
-        if(pictureRef != null && pictureRef.get() != null && !ExecutionMode.getInstance().isTest()){
-            StorageReference ref = getReference().child("images/"+ pictureRef.get());
-            ref.delete();
+    public Task<Void> deleteImageFromStorage(ObservableField<String> pictureRef, String category) {
+
+        if(Storage.checkStoragePath(category) && pictureRef != null && pictureRef.get() != null){
+            String path = category.toLowerCase()+"/";
+            StorageReference ref = getReference().child(path+pictureRef.get());
+            return ref.delete();
+        }
+        return null;
+    }
+
+    @Override
+    public Bitmap getPictureFromDevice(int requestCode, int resultCode, Intent data, Context context, ImageView view) {
+        if (requestCode == GET_FROM_GALLERY && resultCode == Activity.RESULT_OK) {
+            Uri selectedImage = data.getData();
+            Bitmap bitmap = null;
+            try {
+                bitmap = MediaStore.Images.Media.getBitmap(context.getContentResolver(), selectedImage);
+                view.setImageBitmap(bitmap);
+                return bitmap;
+            } catch (FileNotFoundException e) {
+                e.printStackTrace();
```

Comment:
Can't we try to combine the 2 catches that are done here since you are not treating them differently

Reply:
I will have a look !